src/main/java/nextstep/qna/service/QnAService.java

```diff
@@ -26,24 +23,13 @@ public class QnAService {
     @Transactional
     public void deleteQuestion(NsUser loginUser, long questionId) throws CannotDeleteException {
         Question question = questionRepository.findById(questionId).orElseThrow(NotFoundException::new);
-        if (!question.isOwner(loginUser)) {
-            throw new CannotDeleteException("질문을 삭제할 권한이 없습니다.");
-        }
-
-        List<Answer> answers = question.getAnswers();
-        for (Answer answer : answers) {
-            if (!answer.isOwner(loginUser)) {
-                throw new CannotDeleteException("다른 사람이 쓴 답변이 있어 삭제할 수 없습니다.");
-            }
-        }
-
-        List<DeleteHistory> deleteHistories = new ArrayList<>();
-        question.setDeleted(true);
-        deleteHistories.add(new DeleteHistory(ContentType.QUESTION, questionId, question.getWriter(), LocalDateTime.now()));
-        for (Answer answer : answers) {
-            answer.setDeleted(true);
-            deleteHistories.add(new DeleteHistory(ContentType.ANSWER, answer.getId(), answer.getWriter(), LocalDateTime.now()));
-        }
-        deleteHistoryService.saveAll(deleteHistories);
+        DeleteHistories deleteHistories = question.delete(loginUser);
+
+        Answers answers = question.getAnswers();
+        answers.deleteAnswers(loginUser);
```

Comment:
```suggestion
        DeleteHistories deleteHistories = question.delete(loginUser);
```

```java

        Answers answers = question.getAnswers();
        answers.deleteAnswers(loginUser);
```
이 부분도 `question.delete` 안에서 처리할 수 있지 않을까요?

Reply:
```java
DeleteHistories deleteHistories = question.delete(loginUser);
        Answers answers = question.getAnswers();
        answers.deleteAnswers(loginUser);
```

삭제 히스토리 처리와 삭제 처리를 각각 별도 호출을 통해 처리하고 있습니다.
삭제가 되지 않았는데 삭제 히스토리를 만들 수 있다는 것은 어쩌면 이상하지 않을까요?

answers도 삭제와 동시에 삭제 히스토리를 반환하도록 만들었다면 
서비스에서 deleteAnswers를 호출할 필요가 사라질 것 같고 버그 발생도 줄일 수 있을 것 같아요